packages/samples/headless-react/src/pages/AtomicReactPage.tsx

```diff
@@ -0,0 +1,36 @@
+import {
+  AtomicResultList,
+  AtomicSearchBox,
+  AtomicSearchInterface,
+} from '@coveo/atomic-react';
+import {useEffect} from 'react';
+
+export function AtomicReactPage() {
+  useEffect(() => {
+    init();
+  }, []);
+
+  return (
+    <AtomicSearchInterface>
+      <AtomicSearchBox />
+      <AtomicResultList />
+    </AtomicSearchInterface>
+  );
+}
+
+async function init() {
+  await customElements.whenDefined('atomic-search-interface');
```

Comment:
We talked a bit about this this morning. Couple things: 

1- There would be a more idiomatic way to write this in React, with `useRef`, I think. https://reactjs.org/docs/hooks-reference.html#useref

2- In a ideal world, if we could end up with something similar to this:  

```js

const someArbitraryCodeToDetermineWhenToLaunchFirstQuery = ()=> {
 // [...]
}

return (
   <AtomicEngine.Provider provider={myEngine} >
      <AtomicSearchInterface onReady={someArbitraryCodeToDetermineWhenToLaunchFirstQuery} >
        <AtomicSearchBox />
        <AtomicResultList />
      </AtomicSearchInterface>

   </AtomicEngine.Provider>
  );
```

Which means that we would need to both give access to a context provider, and handle that in our react "AtomicSearchInterface", as well as augmenting it to expose a new `onReady` prop.

I'm not sure if that is feasible ? 

Maybe one approach would be to expose a different `AtomicSearchInterface` component in `headless-react` package, that simply wraps the internal "AtomicSearchInterface" that is actually built and shipped with Atomic ?

Reply:
✔️ for point 1

For point 2, this would be a nicer api. We could wrap AtomicSearchInterface to intercept the engine form the provider api and handle calling `initialize` (essentially, the code here would move into the wrapped component). I will try it out,

Reply:
If we go with solution 2, I think it'd be a cherry on top if we had a `useEngine` instead of `useContext`.

Reply:
I encountered an error when using the wrapped component from the atomic-react package.
If the code is inside `headless-react`, it's no problem. Copying the code into `atomic-react` and importing it, I see,

![invalid-hook-call](https://user-images.githubusercontent.com/5565841/146428814-ce477614-7321-4498-8278-225fd2afb68c.PNG)